src/Types.hs

```diff
@@ -77,6 +79,12 @@ type CSI' s a = CSI s s a a
 csi :: CSI s t a b => Iso s t a b
 csi = iso cs cs
 
+-- FIXME: [NP] I'm not quite sure of what this means but at
+-- least this is useful in combination with DF.optionalText
+-- (see Frontend.Page.User) where
```

Comment:
you mean you don't understand it yourself?  in that case i would prefer if you could keep it out of the code, and do something more accessible instead.